pkg/util/kubernetes.go

```diff
@@ -307,3 +308,97 @@ func RecoveryEqual(old, new *api.Recovery) bool {
 	}
 	return reflect.DeepEqual(oldSpec, newSpec)
 }
+
+func CreateRecoveryJob(recovery *api.Recovery, restic *api.Restic, tag string) *batch.Job {
+	job := &batch.Job{
```

Comment:
It should have OwnerRef to the Recovery so that Job is deleted if the user deleted the Recovery object in the middle.